src/context/AppProvider/AppContext.tsx

```diff
@@ -135,82 +135,56 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
+          const { chainNamespace, chainReference } = fromChainId(chainId)
+
+          switch (chainNamespace) {
+            case CHAIN_NAMESPACE.Bitcoin: {
               if (!supportsBTC(wallet)) continue
-              const supportedAccountTypes = (
-                adapter as unknown as bitcoin.ChainAdapter
-              ).getSupportedAccountTypes()
-              for (const accountType of supportedAccountTypes) {
-                const accountParams = utxoAccountParams(chainId, accountType, 0)
-                const { bip44Params, scriptType } = accountParams
+
+              const utxoAdapter = adapter as unknown as UtxoBaseAdapter<UtxoChainId>
+
+              for (const accountType of utxoAdapter.getSupportedAccountTypes()) {
+                const { bip44Params, scriptType } = utxoAccountParams(chainId, accountType, 0)
                 const pubkeys = await wallet.getPublicKeys([
                   {
-                    coin: 'bitcoin',
+                    coin: utxoAdapter.getCoinName(),
                     addressNList: bip32ToAddressNList(toRootDerivationPath(bip44Params)),
                     curve: 'secp256k1',
                     scriptType,
                   },
                 ])
-                if (!pubkeys?.[0]?.xpub) {
-                  throw new Error(`usePubkeys: error getting bitcoin xpub`)
-                }
-                const pubkey = convertXpubVersion(pubkeys[0].xpub, accountType)
 
+                if (!pubkeys?.[0]?.xpub) throw new Error('failed to get public key')
+
+                const pubkey = convertXpubVersion(pubkeys[0].xpub, accountType)
                 if (!pubkey) continue
+
                 acc.push({ [chainId]: pubkey })
               }
               break
             }
-            case dogeChainId: {
-              if (!supportsBTC(wallet)) continue
-              const supportedAccountTypes = (
-                adapter as unknown as dogecoin.ChainAdapter
-              ).getSupportedAccountTypes()
-              for (const accountType of supportedAccountTypes) {
-                const accountParams = utxoAccountParams(chainId, accountType, 0)
-                const { bip44Params, scriptType } = accountParams
-                const addressNList = bip32ToAddressNList(toRootDerivationPath(bip44Params))
-                const pubkeys = await wallet.getPublicKeys([
-                  {
-                    coin: 'dogecoin',
-                    addressNList,
-                    curve: 'secp256k1',
-                    scriptType,
-                  },
-                ])
-                if (!pubkeys?.[0]?.xpub) {
-                  throw new Error(`usePubkeys: error getting dogecoin xpub`)
-                }
-                const pubkey = pubkeys[0].xpub
-                if (!pubkey) continue
-                acc.push({ [chainId]: pubkey })
+
+            case CHAIN_NAMESPACE.Ethereum: {
+              if (chainReference === CHAIN_REFERENCE.EthereumMainnet) {
```

Comment:
Switching on `chainReference` might be nicer as we add more EVMs.

Reply:
I was wanting to either expose a `supportsWallet` function on the adapter or move this check into the relevant functions interacting with the wallet so we can remove this entirely. Would allow us to call any eth or cosmos namespace in the same manner